### ISSUE TITLE
[Feature Request] The official support for kubernetes cluster deployment

### ISSUE THREAD

**Issue by AH-dark** [Anankke/SSPanel-Uim#1987]
**Feature 功能请求**

In the era of rapid development of distributed services, many companies are using clustered solutions. Among them, Kubernetes (K8s) is the most widespread and common clustering solution, and many cloud service providers (such as Google Cloud Platform, AWS, etc.) offer managed services for Kubernetes.

Kubernetes has good maintenance capabilities and scalability, and can adequately handle the problems of server load capacity insufficient due to user growth. PHP applications themselves often have the characteristic of being stateless, which makes them easier to be transformed into clusters.

In summary, I suggest that you adapt SSPanel Uim for clusterization to serve more users.

I have some experience with application clustering and can provide necessary assistance.

**Comment by M1Screw** [comment]
> Kubernetes has good maintenance capabilities and scalability, and can adequately handle the problems of server load capacity insufficient due to user growth.

Or, trying to solve a problem that doesn't exist? 

All these "microservices", "containerization", and "why there is no docker deployment option" questions get me wondering, do these people all have hundreds of bare metal servers to play with? Do these people have ever tried the auto-scaling feature provided by the cloud service provider? Do these people tried to build a HA application with LB and clustered DB before? Or do these people even know what Ansible/Saltstack is?

K8s won't magically make more computing resources available to your service, it simply doesn't make sense for a small to median-size deployment. Unless there is a COO or team leader who wants to "follow the trend" in your company/team.

Does big tech use of containers mean that every single app needs to be converted to a container as well? No.
Does container clustering reduce performance and introduce unnecessary complexity? Yes.
Does your petty little 3 1C/2G VM cluster need to be containerized? Depends on how much free time you have.

**Comment by AH-dark** [comment]
Firstly, I respect your viewpoint as a maintainer, but I do have some differing opinions.

In fact, many people I know who use PHP applications, including SSPanel, are considering adapting for clustering. Many small and medium-sized enterprises (teams) are also beginning to consider architecture deployment based on Kubernetes, at least I know quite a few. Many auto-scaling solutions provided by cloud service providers often require more modifications and adaptations, which is not cost-effective. Many teams that do not use large cloud service providers cannot use the auto-scaling solutions provided by these providers, and Kubernetes is a good alternative.

Indeed, Kubernetes does not provide more resources for computing and even takes up some itself, but what it brings is better maintainability and scalability. Rather than constantly expanding the resources of a monolithic server by changing the original 2C/4G machine to 16C/32G and repeating this cycle, why can't I use multiple servers?

You don't even need to make any changes to the application to have a pretty perfect chart. I'm not sure what you mean by "a problem that does not exist," but I believe that **having a chart is better than nothing**, at least it can solve problems.

I put forth my suggestions with a positive attitude and in standardized language, and I can provide a solution that is more than half complete. However, what I received in return was your harsh words. Your response truly reflects your personal quality.

**Comment by M1Screw** [comment]
> However, what I received in return was your harsh words. Your response truly reflects your personal quality.

Oh no, do I hurt your feelings😆? Is the best argument you got? Maybe learn to fuck off and stop wasting others' time. Fucking snowflakes.
